sdk/test-utils/recorder-new/src/matcher.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createPipelineRequest, HttpClient } from "@azure/core-rest-pipeline";
+import { paths } from "./utils/paths";
+import { RecorderError } from "./utils/utils";
+
+export type Matcher = "HeaderlessMatcher" | "BodilessMatcher";
+
+export async function setMatcher(
+  recorderUrl: string,
+  httpClient: HttpClient,
+  matcher: Matcher,
+  recordingId?: string
+): Promise<void> {
+  const url = `${recorderUrl}${paths.admin}${paths.setMatcher}`;
+
+  const request = createPipelineRequest({ url, method: "POST", allowInsecureConnection: true });
+  request.headers.set("x-abstraction-identifier", matcher);
+  if (recordingId) {
+    request.headers.set("x-recording-id", recordingId);
+  }
+
+  const response = await httpClient.sendRequest(request);
+
+  if (response.status !== 200) {
```

Comment:
A few questions:
1. Is 200 the only acceptable response? What about 201, 204, and other 200ish status code?
2. When we throw the RecorderError do you think it makes sense to add the status code and response body?

Reply:
1. Looking at the [test proxy's code](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs#L86-L102), a 200 response is the only successful response that the test proxy will give currently. But agreed it's better practice to accept any 2xx status code. I'll change that.
1. I agree, that's a good idea. It does look like the proxy will give some [useful info](https://github.com/Azure/azure-sdk-tools/blob/5788ecfd5d34d476b50531ef2dd66c31cabf3818/tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs#L423) when there's an error sometimes.

Reply:
cc @scbedd